sdk/tables/azure-data-tables/tests/test_table_client.py

```diff
@@ -223,30 +224,262 @@ def test_client_with_sas_token(self, tables_storage_account_name, tables_primary
             expiry=datetime.utcnow() + timedelta(hours=1),
         )
 
-        with TableServiceClient(base_url, credential=AzureSasCredential(sas_token)) as client:
+        with TableClient.from_table_url(
+            f"{base_url}/{table_name}", credential=tables_primary_storage_account_key
+        ) as client:
+            table = client.create_table()
+            assert table.name == table_name
+
+        conn_str = f"AccountName={tables_storage_account_name};AccountKey={tables_primary_storage_account_key.named_key.key};EndpointSuffix=core.windows.net"
+        with TableClient.from_connection_string(conn_str, table_name) as client:
```

Comment:
Shouldn't this also have a separate `credential=AzureNamedKeyCredential(...)` passed in?